src/main/java/it/pagopa/pm/gateway/service/CcResumeService.java

```diff
@@ -0,0 +1,196 @@
+            return;
+        }
+        processResume(request, entity, requestId);
+    }
+
+    private void processResume(CreditCardResumeRequest request, PaymentRequestEntity entity, String requestId) {
+        String methodCompleted = request.getMethodCompletd();
+        try {
+            StepZeroRequest stepZeroRequest = objectMapper.readValue(entity.getJsonRequest(), StepZeroRequest.class);
+            stepZeroRequest.setIsFirstPayment(false);
+            MethodCompletedEnum methodCompletedEnum = (StringUtils.isBlank(methodCompleted)) ? null : MethodCompletedEnum.valueOf(methodCompleted);
+            if (entity.getResponseType().equalsIgnoreCase(ThreeDS2ResponseTypeEnum.METHOD.name())) {
+                Map<String, String> params = createRequestForStep1(methodCompletedEnum, entity, stepZeroRequest);
+                executeStep1(params, entity, stepZeroRequest);
+            }
+        } catch (Exception e) {
+            log.error(String.format("error during execution of resume for requestId %s, stackTrace: %s ", requestId, Arrays.toString(e.getStackTrace())));
+        }
+    }
+
+    private Map<String, String> createRequestForStep1(MethodCompletedEnum methodCompletedEnum, PaymentRequestEntity entity, StepZeroRequest stepZeroRequest) throws IOException {
+        String correlationId = entity.getCorrelationId();
+        return vPosRequestUtils.buildStepOneRequestParams(methodCompletedEnum, stepZeroRequest, correlationId);
+    }
+
+    @Async
+    private void executeStep1(Map<String, String> params, PaymentRequestEntity entity, StepZeroRequest request) {
+        try {
+            String requestId = entity.getGuid();
+            log.info("Calling VPOS - Step 1 - for requestId: " + requestId);
+            HttpClientResponse clientResponse = callVPos(params);
+            ThreeDS2Response response = vPosResponseUtils.build3ds2Response(clientResponse.getEntity());
+            vPosResponseUtils.validateResponseMac(response.getTimestamp(), response.getResultCode(), response.getResultMac(), request);
+            if (checkStep1ResultCode(response, entity)) {
+                executeAccount(entity, request);
+            }
+        } catch (Exception e) {
+            log.error(GENERIC_ERROR_MSG + entity.getIdTransaction() + " stackTrace: " + Arrays.toString(e.getStackTrace()));
+        }
+    }
+
+    private HttpClientResponse callVPos(Map<String, String> params) throws IOException {
+        HttpClientResponse clientResponse = httpClient.post(vposUrl, ContentType.APPLICATION_FORM_URLENCODED.getMimeType(), params);
+        if (clientResponse.getStatus() != HttpStatus.OK.value()) {
+            log.error("HTTP Response Status: " + clientResponse.getStatus());
+            throw new IOException("Non-ok response from VPos. HTTP status: " + clientResponse.getStatus());
+        }
+        return clientResponse;
+    }
+
+    private boolean checkStep1ResultCode(ThreeDS2Response response, PaymentRequestEntity entity) {
```

Comment:
It is always better to call boolean methods something like "is..."
For example:
```suggestion
    private boolean isStepOneResultCodeOk(ThreeDS2Response response, PaymentRequestEntity entity) {
```